iml-command-utils/src/lib.rs

```diff
@@ -38,28 +52,95 @@ impl From<Vec<Command>> for CommandError {
     }
 }
 
+pub async fn create_command<T: serde::Serialize>(
+    cmd_body: SendCmd<T>,
+) -> Result<Command, CommandError> {
+    let client = iml_manager_client::get_client()?;
+    let resp = post(client, Command::endpoint_name(), cmd_body)
+        .await?
+        .error_for_status()?;
+
+    let cmd = resp.json().await?;
+
+    tracing::debug!("Resp JSON is {:?}", cmd);
+
+    Ok(cmd)
+}
+
 fn cmd_finished(cmd: &Command) -> bool {
     cmd.complete
 }
 
+pub async fn wait_for_cmd(cmd: Command) -> Result<Command, CommandError> {
+    loop {
+        if cmd_finished(&cmd) {
+            return Ok(cmd);
+        }
+
+        delay_for(Duration::from_millis(1000)).await;
+
+        let client = iml_manager_client::get_client()?;
+
+        let cmd = iml_manager_client::get(
+            client,
+            &format!("command/{}", cmd.id),
+            Vec::<(String, String)>::new(),
+        )
+        .await?;
+
+        if cmd_finished(&cmd) {
+            return Ok(cmd);
+        }
+    }
+}
+
 /// Waits for command completion and prints progress messages
 /// This *does not* error on command failure, it only tracks command
 /// completion
 pub async fn wait_for_cmds(cmds: &[Command]) -> Result<Vec<Command>, CommandError> {
```

Comment:
I'd prefer to have 2 implementations of the function instead of the code interspersed with annotations in such way, sorry.

Reply:
Another approach I can think of is having `wait_for_cmds` be a HOF that takes a callback for each tick. The callback passed could itself be put behind the feature flag.

Reply:
I tried kind of middle ground - there are two implementations hidden inside one using conditional compilation. What do you think?